pkg/provider-local/webhook/prometheus/mutator.go

```diff
@@ -58,7 +58,7 @@ func (m *mutator) Mutate(_ context.Context, newObj, _ client.Object) error {
 				{
 					SourceLabels: []monitoringv1.LabelName{"__name__"},
 					// TODO: add filter for relevant metrics
-					Regex:  "(up)",
+					Regex:  "^(up|task_delay_seconds|task_duration_seconds|total_duration_seconds|task_result_total)$",
```

Comment:
For the test runs on this PR, the regex filtered out the new flow metrics, because they start with `flow_`, see [grafana](https://monitoring-build.prow.gardener.cloud/explore?schemaVersion=1&panes=%7B%2280z%22:%7B%22datasource%22:%22PD1DAB54D242CCAC7%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22count%28%7Bprow_build_id%3D%5C%221867528845205180416%5C%22%7D%29%20by%20%28job,%20__name__%29%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PD1DAB54D242CCAC7%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%221734088234080%22,%22to%22:%221734092357028%22%7D%7D%7D&orgId=1).

Additionally, we can remove the regex anchors. From the [prometheus docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#regex):

> The regex is anchored on both ends. To un-anchor the regex, use `.*<regex>.*`.


```suggestion
					Regex:  "(up|flow_.+)",
```